File: PancakeSorting.cpp
class Solution {
public:
    vector<int> pancakeSort(vector<int>& arr) {
        vector<int> pan;
        int n = arr.size();
        int i = n-1;
        
        while(i >= 0)
        {
            if(i == 0 && arr[i] == 1)
                break;
            
            if(arr[i] == n)
            {
                if((i+1) == arr[i])
                {
                    i--;
                    n--;
                    continue;
                }
                reverse(arr.begin(),arr.begin() + i+1);
                pan.push_back(i+1);
                reverse(arr.begin(),arr.begin()+ n);
                pan.push_back(n);
                n--;
                i = n-1;
            }
            else
                i--;
        }
        
        
        
        return pan;
        
    }
};

File: PancakeSorting.py
class Solution:
    def pancakeSort(self, arr: List[int]) -> List[int]:
        flips = []
        n = len(arr)
        i = len(arr) - 1

        while i >= 0:

            if i == 0 and arr[i] == 1:
                break
            
            if arr[i] == n:

                if arr[i] != i+1:
                    if i != 0:
                        temp = arr[0:i+1]
                        temp.reverse()
                        arr = temp + arr[i+1:]
                        flips.append(i+1)

                    temp = arr[0:n]
                    temp.reverse()
                    arr = temp + arr[n:]
                    flips.append(n)

                n -= 1
                i = n - 1

            else:
                i -= 1

        return flips
